src/passport-saml/saml.ts

```diff
@@ -547,21 +573,21 @@ class SAML {
       ].join('\r\n'));
     };
 
-    this.generateAuthorizeRequest(req, this.options.passive, true, (err, request) => {
+    this.generateAuthorizeRequest(req, this.options.passive, true, (err: Error | null, request?: string) => {
       if (err) {
         return callback(err);
       }
 
       if (this.options.skipRequestCompression) {
-        getAuthorizeFormHelper(null, Buffer.from(request, 'utf8'));
+        getAuthorizeFormHelper(null, Buffer.from(request!, 'utf8'));
```

Comment:
I feel like I'm missing some line of reasoning here in your code. I see lots of `!` being used when I expect `?` to be used. We need to check if these values are `null`, not assert that they aren't. Can you please comment on what I'm missing?

I won't flag the rest of my questions of this nature at this time.